contracts/SpokePool.sol

```diff
@@ -883,12 +959,40 @@ abstract contract SpokePool is
         }
     }
 
+    function _updateCountFromFill(
+        uint256 startingFillAmount,
+        uint256 endingFillAmount,
+        uint256 totalFillAmount,
+        int64 realizedLPFeePct,
+        address token,
+        bool localRepayment,
+        bool useContractFunds
+    ) internal {
+        // If this is a slow fill or it's an initial 0-fill, do nothing, as these should not impact the count.
+        if (useContractFunds || endingFillAmount == 0) return;
+
+        // If this is the first fill and it's partial, assume the rest of the fill will be slow filled (refunded on this chain).
+        if (startingFillAmount == 0 && totalFillAmount - endingFillAmount > 0) {
+            fillCounter[token] += _computeAmountPostFees(totalFillAmount - endingFillAmount, realizedLPFeePct);
+        }
+
+        // If this is not the first fill, remove the partial fill that was previously assumed.
+        if (startingFillAmount != 0) {
```

Comment:
What if there were multiple partial fills that were not the first fill?

Reply:
We're not removing the entire slow fill amount, only the amount that was replaced, so I _think_ this should handle this correctly (but please lmk if you think it's wrong!).